tests/test_logfire.py

```diff
@@ -3221,15 +3222,28 @@ def test_exit_ended_span(exporter: TestExporter):
 )
 def test_default_id_generator(exporter: TestExporter) -> None:
     """Test that SeededRandomIdGenerator generates trace and span ids without errors."""
-    with ExitStack() as stack:
-        for i in range(1024):
+    for i in range(32):
+        with ExitStack() as stack:
             for j in range(8):
                 stack.enter_context(logfire.span(f'span {i}:{j}'))
+                # make sure we get a mix of traces spread across ms
+                # test should take < 32 * 8 ms (0.25s) to run
+                sleep(1 / 1000 / 5)
```

Comment:
Rather inject ns_timestamp_generator into the ID generator, then the exact IDs will be deterministic in tests.

Reply:
and if you really want to generate lots of spans, this is significantly faster:

```python
    for i in range(1024 * 8):
        logfire.info('test', i=i)
    sorted_by_trace_id = [
        export['attributes']['i'] for export in sorted(exported, key=lambda span: span['context']['trace_id'])
    ]
```